### ISSUE TITLE
Paused stream keeps node alive

### ISSUE THREAD

**Issue by piscisaureus** [joyent/node#6653]
Test code:

```js
var conn = require('net').connect(80, 'www.google.com');
// Should not be necessary but just to be sure:
conn.pause();
```

Both node v0.10.x and master hang and don't exit.

@isaacs, any idea? The libuv handle is clearly reading; is node secretly reading under the hood?

**Comment by jasnell** [comment]
@joyent/node-coreteam @piscisaureus ... the hang still happens in v0.10 but not in v0.12 or io.js... do we care? Is there a fix we can / should backport to v0.10?

**Comment by piscisaureus** [comment]
This is a streams question - @cjihrig or @chrisdickinson may know the answer

**Comment by chrisdickinson** [comment]
Yep, looks like I patched it to solve #8200: 4ef2a5a6720c17870e3b91b224fb3dc1e3655d27. Downside is that it adds a new API (`isPaused`.)

**Comment by jasnell** [comment]
Hmm.. I'm inclined just to close then and leave the v0.10 behavior as is.

**Comment by jasnell** [comment]
Closing. This leaves the existing v0.10 behavior alone. Can reopen if folks feel it's necessary to fix this in v0.10
